sentry/src/main/java/io/sentry/hints/Hints.java

```diff
@@ -9,21 +13,51 @@ public final class Hints {
 
   private final @NotNull Map<String, Object> internalStorage = new HashMap<String, Object>();
 
+  public static @NotNull Hints withAttachment(@Nullable Attachment attachment) {
+    @NotNull final Hints hints = new Hints();
+    hints.getAttachmentContainer().add(attachment);
+    return hints;
+  }
+
+  public static @NotNull Hints withAttachments(@Nullable List<Attachment> attachments) {
+    @NotNull final Hints hints = new Hints();
+    hints.getAttachmentContainer().addAll(attachments);
+    return hints;
+  }
+
   public void set(@NotNull String hintType, @Nullable Object hint) {
     internalStorage.put(hintType, hint);
   }
 
-  public @Nullable Object get(@NotNull String hintType) {
-    return internalStorage.get(hintType);
+  public @Nullable Object get(@NotNull String hintName) {
+    return internalStorage.get(hintName);
   }
 
-  // TODO maybe not public
-  public void remove(@NotNull String hintType) {
-    internalStorage.remove(hintType);
+  @SuppressWarnings("unchecked")
+  public <T extends Object> @Nullable T getAs(@NotNull String hintName, @NotNull Class<T> clazz) {
+    Object hintValue = internalStorage.get(hintName);
+
+    if (clazz.isInstance(hintValue)) {
+      return (T) hintValue;
+    } else {
+      return null;
+    }
+  }
+
+  public void remove(@NotNull String hintName) {
+    internalStorage.remove(hintName);
   }
 
-  // TODO addAttachment(one)
```

Comment:
`h`: I'm not sure if we should add the attachments container.

Option A
```Kotlin
val hint = Hints()
hint.set("my-hint", "my-value")
hint.attachmentsContainer.add(attachment)

// improved version
hint.attachments.add(attachment)
```
VS

Option B
```Kotlin
val hint = Hints()
hint.set("my-hint", "my-value")
hint.addAttachment(attachment)
```

I like B more because it's shorter and simpler. What do you think @mattjohnsonpint, @bruno-garcia, @marandaneto and @romtsn?

Reply:
I initially built it like `Option B` which caused a lot more weird handling of casting, checking for non null etc. due to Map<String, Object>. I just wanted to avoid this source of bugs. Also it felt like I was bloating `Hints` with just attachments alone. Not sure how much we plan on adding there in the future. By keeping this separate it may be easier for devs to figure out what they can do with attachments.

Reply:
Created https://github.com/getsentry/sentry-java/pull/2049 to show what this might look like. I prefer the version in this PR here over the other one as it feels a lot cleaner and doesn't have to worry about type erasure. Is the indirection really this much of a pain for developers? I'd rather maintain this than the other version 😅 if there's no big disadvantage for developers.

Reply:
> I initially built it like Option B which caused a lot more weird handling of casting, checking for non null etc. due to Map<String, Object>.

Yes, this would be worse than B. For me, B only works if we use the `AttachmentsContainer`, which we should rename to `Attachment` and make it an inner class of Hint. Then you don't have to do all the casting.

Reply:
I've already renamed to `Attachments`. I can't see what difference it would make to move it to an inner class. I've seen lots of times that things get bloated with inner classes, which is better than no separation, but still harder to go through than separate class files. Or do you mean `Attachments` should become a property of `Hints` instead of being stored in the internal map?

Reply:
Changed it to a separate property on `Hints` and moved methods to `Hints` now.

Reply:
> Changed it to a separate property on Hints and moved methods to Hints now.

Even better. Then we don't need to have this discussion.